src/lib/math/pcurves/pcurves_impl/pcurves_impl.h

```diff
@@ -698,14 +705,48 @@ class ProjectiveCurvePoint {
       }
 
       constexpr Self dbl_n(size_t n) const {
-         // TODO it is possible to optimize this by carrying over values from
-         // the previous iteration into the next
+         if constexpr(Self::A_is_minus_3) {
```

Comment:
In case you want to express this as a good-old function overload, I think you could do this:

```C++
constexpr Self dbl_n(size_t n const)
   requires(Self::A_is_minus_3)
{
   // ...
}

constexpr Self dbl_n(size_t n const)
   requires(!Self::A_is_minus_3)
{
   // ...
}
```

... to avoid the `if constexpr` spanning the entirety of the function. No strong opinion on what is more idiomatic from my side

Reply:
I expect in the future this will expand to cover other types of curves (A == 0 and/or generic A) in which case the single function will be cleaner.